src/app/static/src/app/generated.d.ts

```diff
@@ -400,6 +528,10 @@ export interface ModelCalibrateRequest {
 export type ModelCancelResponse = null;
 export interface ModelOptionsValidate {
   valid: true;
+  warnings: {
+    text: string;
+    locations: ("model_options" | "model_fit" | "model_calibrate" | "review_output" | "download_results")[];
+  }[];
```

Comment:
I'm learning new ts here. I guess this signifies an array of locations, containing one or more of the above string values?